### PR TITLE
Add Option to loop through League

### PR DESCRIPTION
**Pull request by andrew951**
Looking to resolve Issue #68 
Created logic to check if 5 is selected, then loops through a variable stored in localStorage from 0 to 4. If 5, then resets back to 0.

### CODE REVIEW

File: enhancedautoclicker.js [hunk 1]
```diff
@@ -267,16 +269,27 @@ function autoGym() {
...
-                            break
+                //If "All" is selected, then go through list of league from 0-4
+                if(gymSelect === 5) {
+                    var allSelectedGym = localStorage.getItem('allSelectedGym')
+                    GymRunner.startGym(player.town().content[allSelectedGym])
+                    allSelectedGym++
+                    if(allSelectedGym === 5) {
+                        allSelectedGym = 0
+                    }
+                    localStorage.setItem("allSelectedGym", allSelectedGym);
```
Comment (KarmaAlex) [inline enhancedautoclicker.js:280]: localStorage should only be used for variables you need to remember between sessions but this one is reset on startup, consider using a local variable instead. Reason why is localStorage has a limit of variables you can save locally so it should be used sparsely.

**Review by KarmaAlex — CHANGES_REQUESTED** [review]
Avoid using localStorage in this case

File: enhancedautoclicker.js (hunk 1 above)
Comment (andrew951) [reply enhancedautoclicker.js]: Updated to use variable. Haven't tested it

**Comment by iluno90** [comment]
Something I noticed while testing this is that under some circumstances, it can cause the battle screen to glitch/become empty, and the map area to disappear completely.

Image

For example, if you are doing the all option on an Elite 4, and then click the docks and move to a different region, it will usually happen. Also, if you go to a regular gym and run the auto gym on all, it will glitch out after the gym is cleared once.

**Comment by andrew951** [comment]
I think I know why. I can update later today 

> Something I noticed while testing this is that under some circumstances, it can cause the battle screen to glitch/become empty, and the map area to disappear completely.
> 
> Image
> 
> For example, if you are doing the all option on an Elite 4, and then click the docks and move to a different region (without first disabling auto gym), it will usually happen. Also, if you go to a regular gym and run the auto gym on all, it will glitch out after the gym is cleared once.

**Comment by andrew951** [comment]
> Something I noticed while testing this is that under some circumstances, it can cause the battle screen to glitch/become empty, and the map area to disappear completely.
> 
> Image
> 
> For example, if you are doing the all option on an Elite 4, and then click the docks and move to a different region (without first disabling auto gym), it will usually happen. Also, if you go to a regular gym and run the auto gym on all, it will glitch out after the gym is cleared once.

Should be resolved now! :)

File: enhancedautoclicker.js [hunk 2]
```diff
@@ -270,7 +270,12 @@ function autoGym() {
 
             if (App.game.gameState != GameConstants.GameState.gym) {
                 //If "All" is selected, then go through list of league from 0-4
-                if(gymSelect === 5) {
+                if(gymSelect === 5 && (((player.town().content[0] instanceof Gym && player.town().content[0].isUnlocked()) &&
```
Comment (KarmaAlex) [inline enhancedautoclicker.js:273]: You don't need to check if they are all unlocked, you can just check 

`if (gymSelect === 5 && (player.town().content[gymSelect] instanceof Gym && player.town().content[gymSelect].isUnlocked()) || (player.town().content[gymSelect] instanceof Champion && player.town().content[gymSelect].isUnlocked()))`

**Review by KarmaAlex — CHANGES_REQUESTED** [review]
Seems to work well, the only issue you may want to fix is that when turning it off after clearing, say, the second elite 4 member next time you start a new run it will start from the third member.

File: enhancedautoclicker.js (hunk 2 above)
Comment (andrew951) [reply enhancedautoclicker.js]: Updated my patch to include this instead

**Pull request merged**